src/test/java/org/opensearch/security/http/HTTPOnBehalfOfJwtAuthenticatorTest.java

```diff
@@ -0,0 +1,287 @@
+                "",
+                claimsEncryptionKey,
+                Jwts.builder().setSubject("Leonard McCoy"),
+                false);
+
+        Assert.assertNull(credentials);
+    }
+
+    @Test
+    public void testBadKey() throws Exception {
+
+        final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                BaseEncoding.base64().encode(new byte[]{1,3,3,4,3,6,7,8,3,10}),
+                claimsEncryptionKey,
+                Jwts.builder().setSubject("Leonard McCoy"),
+                false);
+
+        Assert.assertNull(credentials);
+    }
+
+    @Test
+    public void testTokenMissing() throws Exception {
+
+        HTTPOnBehalfOfJwtAuthenticator jwtAuth = new HTTPOnBehalfOfJwtAuthenticator(BaseEncoding.base64().encode(secretKeyBytes),claimsEncryptionKey);
+        Map<String, String> headers = new HashMap<String, String>();
+
+        AuthCredentials credentials = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<String, String>()), null);
+
+        Assert.assertNull(credentials);
+    }
+
+    @Test
+    public void testInvalid() throws Exception {
+
+        String jwsToken = "123invalidtoken..";
+
+        HTTPOnBehalfOfJwtAuthenticator jwtAuth = new HTTPOnBehalfOfJwtAuthenticator(BaseEncoding.base64().encode(secretKeyBytes), claimsEncryptionKey);
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put("Authorization", "Bearer "+jwsToken);
+
+        AuthCredentials credentials = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<String, String>()), null);
+        Assert.assertNull(credentials);
+    }
+
+    @Test
+    public void testBearer() throws Exception {
+
+        String jwsToken = Jwts.builder().setSubject("Leonard McCoy").setAudience("ext_0").signWith(secretKey, SignatureAlgorithm.HS512).compact();
+
+        HTTPOnBehalfOfJwtAuthenticator jwtAuth = new HTTPOnBehalfOfJwtAuthenticator(BaseEncoding.base64().encode(secretKeyBytes), claimsEncryptionKey);
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put("Authorization", "Bearer "+jwsToken);
+
+        AuthCredentials credentials = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<String, String>()), null);
+
+        Assert.assertNotNull(credentials);
+        Assert.assertEquals("Leonard McCoy", credentials.getUsername());
+        Assert.assertEquals(0, credentials.getBackendRoles().size());
+        Assert.assertEquals(2, credentials.getAttributes().size());
+    }
+
+    @Test
+    public void testBearerWrongPosition() throws Exception {
+
+        Settings settings = Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKeyBytes)).build();
+
+        String jwsToken = Jwts.builder().setSubject("Leonard McCoy").signWith(secretKey, SignatureAlgorithm.HS512).compact();
+
+        HTTPJwtAuthenticator jwtAuth = new HTTPJwtAuthenticator(settings, null);
```

Comment:
This will be fixed on the next sync up with main branch. It should be change to the testing case of `HTTPOnBehalfOfAuthenticator()` for testing.